app/models/form_item.rb

```diff
@@ -17,8 +17,8 @@ class FormItem < ApplicationRecord
 
   # These associations have qing in their foreign keys but we have them here in FormItem instead
   # because we will eventually support conditions on groups.
-  has_many :display_conditions, -> { with_display_role }, class_name: "Condition",
-    foreign_key: :questioning_id, dependent: :destroy, inverse_of: :questioning
+  has_many :display_conditions, -> { with_display_role.by_ref_qing_rank }, class_name: "Condition",
+    foreign_key: :questioning_id, dependent: :destroy, inverse_of: :questioning, autosave: false
```

Comment:
Why autosave false? I don't think I've ever seen this. Isn't false the default? If not, Could you maybe add a comment about why this is needed?

Reply:
I thought that it was needed for dirty tracking to work properly on that association, but I appear to have been mistaken, so I removed it.